### PR TITLE
Add WiFiSSLClient

### PR DESCRIPTION
**Pull request by copercini**
Provide SSL/TLS functions to ESP32 with Arduino IDE

### CODE REVIEW

**Comment by me-no-dev** [comment]
let's start with moving the files out of the core. Maybe the whole SSL Client and eventually server should be a separate library (again nothing in the core).
Then I'll run some formatting on the code and we will look from there :) how does that sound?

**Comment by copercini** [comment]
Sounds good. 

The files in the core can easily be moved to the same folder of library.  Also,  you can format or edit anything you want =)

File: libraries/WiFi/src/ssl_drv.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* Provide SSL/TLS functions to ESP32 with Arduino IDE
+ * Ported from Ameba8195 (no license specified) to ESP32 by Evandro Copercini - 2017 - Public domain
```
Comment (igrr) [inline libraries/WiFi/src/ssl_drv.h]: I'm afraid we need to get some confirmation about the original license of these files before we can consider adding them to the project.

Comment (copercini) [inline libraries/WiFi/src/ssl_drv.h]: Yes, you are right.

These files process the input to make a good call to mod_ssl.c

I will rewrite them based on WifiClientSecure of ESP8266 and make this lib compatible with ESP8266 arduino sketches.

Also I will move out of the core folder as @me-no-dev request.

**Pull request closed without merging**